iosApp/iosApp/Pages/NearbyTransit/NearbyViewModel.swift

```diff
@@ -20,12 +22,28 @@ class NearbyViewModel: ObservableObject {
     }
 
     @Published var departures: StopDetailsDepartures?
-    @Published var navigationStack: [SheetNavigationStackEntry] = []
+    @Published var navigationStack: [SheetNavigationStackEntry] = [] {
+        didSet { Task {
+            let navEntry = navigationStack.lastSafe()
+            do {
+                switch navEntry {
+                case let .stopDetails(stop, _):
+                    try await visitHistoryUsecase.addVisit(visit: .StopVisit(stopId: stop.id))
+                default: break
+                }
+            } catch {
+                let navLabel = navEntry.sheetItemIdentifiable().debugDescription
```

Comment:
Will this make the raw error description visible to the user?

Reply:
Oh, nope this is just a warn log for us, nothing that would be visible to users. Unless they tried really hard to read their system logs lol